src/components/CDPDisplay/index.js

```diff
@@ -50,8 +70,18 @@ function CDPView({ cdpId }) {
   ]);
 
   useEffect(() => {
-    trackCdpById(maker, cdpId, dispatch);
-  }, [cdpId, dispatch, maker]);
+    (async () => {
+      const ilk = await maker.service('mcd:cdpManager').getIlkByCdpId(cdpId);
+      const type = cdpTypes.find(f => f.key === ilk);
+      if (type) {
+        trackCdpById(maker, cdpId, dispatch).catch(() =>
```

Comment:
`trackCdpById` throws an error if we try to fetch a vault with an unsupported ilk type. It seemed we didn't have a built in way to discover the ilk type before the error get's thrown in the SDK, so I added a method to the `cdpManager` service.

Reply:
how about adding a try/catch inside trackCdpById instead? that would avoid causing an extra network request for every CDP load. 

actually, i think the `.catch` you've attached to trackCdpById here should be sufficient by itself?